assets/js/components/PostSearcherAutoSuggest.js

```diff
@@ -0,0 +1,112 @@
+/**
+ * PostSearcherAutoSuggest component.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import {
+	Combobox,
+	ComboboxInput,
+	ComboboxPopover,
+	ComboboxList,
+	ComboboxOption,
+} from '@reach/combobox';
+
+/**
+ * WordPress dependencies
+ */
+import { useState, useEffect, useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import { useDebouncedState } from '../hooks/useDebouncedState';
+
+const PostSearcherAutoSuggest = ( { setCanSubmit, setMatch } ) => {
+	const [ searchTerm, setSearchTerm ] = useState( '' );
+	const debouncedValue = useDebouncedState( searchTerm, 200 );
+	const [ results, setResults ] = useState( [] );
+	const noResultsMessage = __( 'No results found', 'google-site-kit' );
+
+	const postSearch = async ( query ) => {
+		try {
+			const queryResults = await API.get( 'core', 'search', 'post-search',
+				{ query: encodeURIComponent( query ) },
+				{ useCache: false }
+			);
+
+			if ( Array.isArray( queryResults ) ) {
+				setResults( queryResults );
+			} else {
+				setResults( [] );
+			}
+		} catch ( err ) {
+			setResults( [] );
+		}
+	};
+
+	const onSelectCallback = useCallback( ( value ) => {
+		if ( Array.isArray( results ) && value !== noResultsMessage ) {
+			const foundMatch = results.find( ( post ) => post.post_title === value );
+			if ( foundMatch ) {
+				setCanSubmit( true );
+				setMatch( foundMatch );
+			}
+		} else {
+			setCanSubmit( false );
+		}
+	}, [ results ] );
+
+	const onInputChange = useCallback( ( event ) => {
+		setCanSubmit( false );
+		setSearchTerm( event.target.value );
+	}, [] );
+
+	useEffect( () => {
+		if ( debouncedValue !== '' ) {
+			postSearch( debouncedValue );
```

Comment:
Since this is the only place `postSearch` is used, [it is recommended to move the definition of this function inside the effect callback](https://reactjs.org/docs/hooks-faq.html#is-it-safe-to-omit-functions-from-the-list-of-dependencies) to better guard against bugs from this function using outside dependencies which aren't declared in the list below.

It doesn't make a difference in this case (yet), but is the safer recommended approach.

Reply:
Reworked this useEffect callback to set API request directly instead of calling postSearch function.